android/src/main/java/com/oney/WebRTCModule/CameraCaptureController.java

```diff
@@ -85,34 +116,48 @@ public void switchCamera(SwitchCameraHandler handler) {
             CameraVideoCapturer capturer = (CameraVideoCapturer) videoCapturer;
             String[] deviceNames = cameraEnumerator.getDeviceNames();
             int deviceCount = deviceNames.length;
+            Log.d(TAG, "Invoked switchCamera, deviceCount:" + deviceCount);
 
             // Nothing to switch to.
             if (deviceCount < 2) {
                 handler.onSwitchCameraDone(facingMode());
                 return;
             }
 
+            CameraVideoCapturer.CameraSwitchHandler cameraSwitchHandler = new CameraVideoCapturer.CameraSwitchHandler() {
+                @Override
+                public void onCameraSwitchDone(boolean b) {
+                    Log.d(TAG, "Invoked onCameraSwitchDone, isFrontFacing:" + b);
+                    isFrontFacing = b;
+                    handler.onSwitchCameraDone(facingMode());
+                }
+
+                @Override
+                public void onCameraSwitchError(String s) {
+                    Log.e(TAG, "Error switching camera: " + s);
+                    handler.onSwitchCameraDone(facingMode());
+                }
+            };
             // The usual case.
             if (deviceCount == 2) {
-                capturer.switchCamera(new CameraVideoCapturer.CameraSwitchHandler() {
-                    @Override
-                    public void onCameraSwitchDone(boolean b) {
-                        isFrontFacing = b;
-                        handler.onSwitchCameraDone(facingMode());
-                    }
-
-                    @Override
-                    public void onCameraSwitchError(String s) {
-                        Log.e(TAG, "Error switching camera: " + s);
-                        handler.onSwitchCameraDone(facingMode());
-                    }
-                });
-                return;
+                capturer.switchCamera(cameraSwitchHandler);
+            } else {
+                try{
+                    // We are always selecting the first one that matches the
+                    // opposite facing mode that we currently have.
+                    String deviceToSwitchTo = Arrays.stream(deviceNames).filter((deviceName) -> {
+                        Log.d(TAG, "Filtering " + deviceName + " isFrontFacing " + cameraEnumerator.isFrontFacing(deviceName));
+                        return cameraEnumerator.isFrontFacing(deviceName) != isFrontFacing;
+                    }).findFirst().get();
+                    Log.d(TAG, "Invoked switchCamera, has more than 02 cameras, will switch to device:" + deviceToSwitchTo);
+                    capturer.switchCamera(cameraSwitchHandler, deviceToSwitchTo);
+                }catch (Exception e) {
+                    Log.d(TAG, "Failed to switch the camera device:" + e.getMessage());
+                }
+                // If we are here the device has more than 2 cameras. Cycle through them
```

Comment:
nit: this comment should be before the above `try` right?

Reply:
The comment was from the original method, which is also commented below:
`// switchCamera(!isFrontFacing, deviceCount, handler);`